tmpl/blog/mirage-entropy.md

```diff
@@ -0,0 +1,362 @@
+# Organized chaos: managing randomness
+
+This post gives a bit of background on the Random Number Generator (RNG) in the
+new MirageOS 2.5.0.
+
+First we give background about why RNGs are really critical for security, then
+we try to clarify the often-confused concepts of "randomness" and "entropy" as
+used in this context, and finally we explore the challenges of harvesting
+good-quality entropy in a unikernel environment.
+
+## The race to... wherever
+
+Security software needs to play dice.
+
+It needs to do so to create secrets, for example. Secrets can then serve as the
+keys that protect communication, like the Diffie-Hellman key that two TLS
+endpoints can exchange. Proof of the knowledge of a particular secret can be
+used to verify the identity of someone on the Internet, as in the case of
+verifying the possession of the secret RSA key associated with an X.509
+certificate. As guessing a secret can have disastrous consequences, it must be
+chosen in a manner that is not realistically predictable to anyone else.
+
+There are other reasons to use randomness. A number of algorithms require a
+unique value every time they are invoked and badly malfunction when this
+assumption is violated, with random choice being one way to provide a value likely
+to be unique. For example, repeating the [`k`-parameter][dsa-sensitivity] in DSA
+digital signatures compromises the secret key, while reusing a GCM
+[nonce][gcm-security] negates authenticity. Other algorithms are probabilistic,
+in that they generate random values before operating on an input, and then store
+the chosen values in the output, such as the [OAEP][wiki-oaep] padding mode for
+RSA. This is done in order to confuse the relationship between the input and
+the output and defeat a clever attacker who tries to manipulate the input to
+gain knowledge about secrets by looking at the output. Yet other algorithms pick
```

Comment:
Yet other -> **Still** other
(mostly style rather than correctness that one though I think :)